File: Java_Practice/algorithm/src/타겟_넘버.java
public class 타겟_넘버 {
    static int answer = 0;
    public void dfs(int[] numbers, int target, int idx, int sum) {
        if (idx == numbers.length) {
            if (sum == target) {
                answer ++;
            }
        } else {
            dfs(numbers, target, idx + 1, sum + numbers[idx]);
            dfs(numbers, target, idx + 1, sum - numbers[idx]);
        }
    }
    public int solution(int[] numbers, int target) {
        dfs(numbers, target, 0, 0);
        return answer;
    }
}

File: DFS,BFS/타겟 넘버.py
def dfs(numbers, target, idx, sum):
    global answer
    if idx == len(numbers):
        if sum == target:
            answer += 1
    else:
        dfs(numbers, target, idx + 1, sum + numbers[idx])
        dfs(numbers, target, idx + 1, sum - numbers[idx])

def solution(numbers, target):
    global answer
    answer = 0

    dfs(numbers, target, 0, 0)

    return answer

print(solution([1, 1, 1, 1, 1], 3))